File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/client/messages/ClientMessageHandshake.java
package net.stzups.board.server.websocket.protocol.client.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.server.websocket.protocol.client.ClientMessage;
import net.stzups.board.server.websocket.protocol.client.ClientMessageType;

public class ClientMessageHandshake extends ClientMessage {
    private long id;
    private long token;
    private String code;

    public ClientMessageHandshake(ByteBuf byteBuf) {
        super(ClientMessageType.HANDSHAKE);
        this.id = byteBuf.readLong();
        this.token = byteBuf.readLong();
        this.code = readString(byteBuf);
    }

    public long getId() {
        return id;
    }

    public long getToken() {
        return token;
    }

    public String getCode() {
        return code;
    }
}

File: frontend/html/scripts/protocol/client/messages/ClientMessageHandshake.js
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";

export default class ClientMessageHandshake extends ClientMessage {
    constructor(token) {
        super(ClientMessageType.HANDSHAKE);
        this.token = token;
    }

    getBufferSize() {
        return super.getBufferSize() + 8;
    }

    serialize(writer) {
        super.serialize(writer);

        writer.writeBigInt64(this.token)
    }
}